Sources/SystemInternals/WindowsSyscallAdapters.swift

```diff
@@ -9,16 +9,19 @@
 
 #if os(Windows)
 
+import ucrt
+import WinSDK
+
 @inline(__always)
-internal func open(_ path: UnsafePointer<CChar>, _ oflag: Int32) {
+internal func open(_ path: UnsafePointer<CChar>, _ oflag: Int32) -> CInt {
   var fh: CInt = -1
   _ = _sopen_s(&fh, path, oflag, _SH_DENYNO, _S_IREAD | _S_IWRITE)
   return fh
 }
 
 @inline(__always)
 internal func open(
-  _ path: UnsafePointer<CChar>, _ oflag: Int32, _ mode: mode_t
+  _ path: UnsafePointer<CChar>, _ oflag: Int32, _ mode: CInt
```

Comment:
I think we should keep the `mode_t` in this interface and do the numeric cast in the body. If Windows doesn't have `mode_t`, then we can have a public typealias for `CModeT` in `Exports.swift`. I was phrasing the `system_foo` functions as using the non-typealiased decls from Darwin/GlibC, but we can change that too if it makes sense.

Reply:
Oh, if we're introducing a CModeT type alias, that's perfect.